src/app/view/timestamp-editor/timestamp-editor.component.ts

```diff
@@ -113,6 +106,11 @@ export class TimestampEditorComponent implements OnInit {
       publisherRole: new FormControl(null),
       vesselDraft: new FormControl(null, [Validators.pattern('^[0-9]+(.[0-9]?)?$')]),
     });
+
```

Comment:
The race was that `timestampFormGroup` was `null`/`undefined` when `setFullVesselDetails` was called.  By setting `timestampFormGroup` because initiating the flow that eventually triggers the offending `setFullVesselDetails` would solve it in my book.